go/vt/vtgate/executor.go

```diff
@@ -1481,7 +1490,7 @@ func getTabletThrottlerStatus(tabletHostPort string) (string, error) {
 	client := http.Client{
 		Timeout: 100 * time.Millisecond,
 	}
-	resp, err := client.Get(fmt.Sprintf("http://%s/throttler/check?app=vtgate", tabletHostPort))
+	resp, err := client.Get(fmt.Sprintf("http://%s/throttler/check-self", tabletHostPort))
```

Comment:
Two things here:
1. You can & should use the new `CheckThrottler` gRPC call, instead of using a http client.
2. If I understand correctly, this code assumes that the throttler's metric is necessarily replication lag. But this is not the case. The throttler could be told to measure anything else, like `Threads_running`, or others. So looking to the throttler is unfortunately unreliable in this context.
  I have it in my plans to make the  throttler measure multiple metrics, one of which will always be `lag`. But it's gonna take a while to get there.

Reply:
I'm actually not sure if this were indeed intended to measure replication lag or not. If not, then:
- Dismiss my review
- Replace HTTP with gRPC
- Do provide some `AppName` - feel free to add a new formal one, whether it's just `vtgate` or something more refined, in https://github.com/vitessio/vitess/blob/d8f771c695b5b82ec8b104b66587018b6e18f2fa/go/vt/vttablet/tabletserver/throttle/throttlerapp/app.go#L43-L65

Reply:
Thanks! I made these changes here: https://github.com/vitessio/vitess/pull/15348/commits/6d07ac271f4e1a8d852a6d5b165c31039a5b8255